CSSS/Kernel.cs

```diff
@@ -172,28 +167,34 @@ public bool PerformTasks()
             }
 
             // If we should shutdown
-            if (shouldShutdown)
+            if (config.CSSSProgramMode.HasFlag(Config.CSSSModes.Shutdown))
             {
                 // Run the shutdown command
                 var process = new System.Diagnostics.Process();
                 process.StartInfo.FileName = "shutdown";
                 switch (config.operatingSystemType)
                 {
                     case Config.OperatingSystemType.WinNT:
-                        // On Windows, we can specify the seconds
-                        // to delay the shutdown :D
-                        // So we shutdown in 15 seconds (should be suffficient time)
-                        process.StartInfo.Arguments = "-s -t 15";
+                        logger.Info("Operating System detected as WinNT - shutting down computer in 5 seconds");
+                        // On Windows, we can specify the seconds to delay the shutdown
+                        // So we shutdown in 15 seconds 
+                        // The logic behind this is that shutting down the program immediately
+                        // may cause issues if there is a longer lasting process and allows
+                        // time for the program to gracefully exit without being terminated
+                        // as the computer shutsdown
+                        process.StartInfo.Arguments = "-s -t 5";
                         break;
                     case Config.OperatingSystemType.Linux:
-                        // Linux doesn't let you specify seconds... >:(
+                        logger.Info("Operating System detected as WinNT - shutting down computer in 1 minute");
+                        // Linux doesn't let you specify seconds so it defaults to 1 minute
+                        // See above command for logic on shutdown
                         process.StartInfo.Arguments = "-h +1";
                         break;
                     default:
-                        logger.Warn("Unknown OS specified in Shutdown - this shouldn't happen");
+                        logger.Warn("Unknown OS specified in Shutdown - you will need to do this manually");

```

Comment:
We don't get the user to specify an OS when they issue `--shutdown`, so saying it isn't specified could be confusing. Can the start of the line read `Unable to shutdown computer` instead?